stag-library/src/main/java/com/vimeo/stag/KnownTypeAdapters.java

```diff
@@ -336,27 +338,36 @@ public static void write(JsonWriter out, char value) throws IOException {
         }
     }
 
-
     /**
      * Helper class for {@link boolean}.
      */
     public static final class PrimitiveBooleanTypeAdapter {
 
+        private static final List<String> VALID_BOOLEAN_AS_STRING = Arrays.asList("true", "false", "1", "0");
+
         public static boolean read(JsonReader in, boolean defaultValue) throws IOException {
             JsonToken peek = in.peek();
             if (peek == JsonToken.NULL) {
                 in.nextNull();
                 return defaultValue;
-            } else if (peek == JsonToken.STRING) {
+            } else if (peek == JsonToken.STRING || peek == JsonToken.NUMBER) {
                 // support strings for compatibility with GSON 1.7
-                return Boolean.parseBoolean(in.nextString());
+                return Boolean.parseBoolean(booleanAsString(in.nextString()));
```

Comment:
I have concerns with throwing exception here. I can't corelate this to a use case that we have but it might break things for some someone where they were returning true along with other values which do not exist in the set of valid boolean values. Anyone doing this would not be doing the right thing but we will deviate from the Gson behavior here.